extensions/api/data-management/contractdefinition/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/model/ContractDefinitionDto.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+
+import java.util.List;
+
+@JsonDeserialize(builder = ContractDefinitionDto.Builder.class)
+public class ContractDefinitionDto {
+    private String accessPolicyId;
+    private String contractPolicyId;
+    private List<Criterion> criteria;
+    private String id;
+
+    private ContractDefinitionDto() {
```

Comment:
Another question: wouldn't it make sense to have a constructer which takes the "normale" object as a parameter and then creates a new DTO object out of it? It's then lot easier to transform a normal object into an DTO.

I your case:
```java
public ContractDefinitionDto(ContractDefinition contractDefinition) {
    accessPolicyId = contractDefinition.getAccessPolicyId();
    ...
} 
```

Reply:
No, DTOs should not accept the domain object in constructors, because then the DTO has a runtime dependency onto `:spi`. The way we will tackle this is by using transformers, that convert DTOs into domain objects and vice versa.

Reply:
Ahh okay and we would like to avoid this in reason of? (Sorry for the noob question)

Reply:
Couple of reasons: the conversion might need additional validation (especially from DTO->BO), there might be additional dependencies that the transformer requires, e.g. when the DTO is actually a composition of several BOs, transformers may be injectable, etc. 
pick your favourite :)

Reply:
The question was misleading - sorry! Transformers makes absolute sense for me. The question was regarding the dependency onto `:spi`

Reply:
it increases decoupling, also DTOs don't need to know about "their" BO.

Reply:
Ahhh okay, thought there would be a bigger reason for that which I don't know. Thanks for the explanation! :)

Reply:
I'd say clean architecture is big enough of a reason :)